### PR TITLE
Watches changes on hd1 and adds new files to hd2 - issue 325

### PR DESCRIPTION
**Pull request by ggrig**
This implements new pre-procesinng APIs that start and stop the watcher thread.
Once the watcher is started no other pre-porcessing API is allowed until the watcher is stopped.
The watcher does not interfere HD2 processing API
#325

### CODE REVIEW

File: cdr_plugin_folder_to_folder/pre_processing/Pre_Processor.py [hunk 1]
```diff
@@ -214,6 +267,39 @@ def process_downloaded_zip_file(self, url):
...
+    def StartHD1WatcherThread(self):
+        if self.ThreadRunning():
+            return
+
+        self.status.StartStatusThread()
+        self.status.set_phase_1()
+
+        self.pre_processing_thread_status = Pre_Processor.THREAD_RUNNIG
+        self.pre_processing_thread = threading.Thread(target=self.PreProcessingThread, args=(10,))
+        self.pre_processing_thread.start()
```
Comment (DinisCruz) [inline cdr_plugin_folder_to_folder/pre_processing/Pre_Processor.py:288]: We need to check the performance implications of this

what happens when there are tons of files in HD1?

We also need a diagram for this change of workflow

**Review by DinisCruz — CHANGES_REQUESTED** [review]
We need to do a set of performance tests on this and document the workflow changes

We also need a detailed test plan to test the changes in behaviour

**Comment by ggrig** [comment]
The implications on the system performance of adding the watcher have been studied. When running on t3.xlarge EC2 instance, the feature does not add load to CPU, RAM, or phase 2 processing speed. The highest load noticed with the watcher thread running (and no phase 2 in progress) was 12.1 % on one of the CPU cores. Mostly it stays in the range from 0% to 8.1%

image

**Comment by ggrig** [comment]
image

**Pull request merged**